src/mono/wasm/runtime/run.ts

```diff
@@ -35,33 +32,35 @@ export async function mono_run_main_and_exit(main_assembly_name: string, args?:
  * Possible signatures are described here  https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/program-structure/main-command-line
  */
 export async function mono_run_main(main_assembly_name: string, args?: string[]): Promise<number> {
-    if (args === undefined || args === null) {
-        args = runtimeHelpers.config.applicationArguments;
-    }
-    if (args === undefined || args === null) {
-        if (ENVIRONMENT_IS_NODE) {
-            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-            // @ts-ignore:
-            const process = await import(/*! webpackIgnore: true */"process");
-            args = process.argv.slice(2) as string[];
-        } else {
-            args = [];
-        }
-    }
+    // TODO MF: Fix mono_run_main
+    // if (args === undefined || args === null) {
+    //     args = runtimeHelpers.config.applicationArguments;
+    // }
+    // if (args === undefined || args === null) {
+    //     if (ENVIRONMENT_IS_NODE) {
+    //         // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    //         // @ts-ignore:
+    //         const process = await import(/*! webpackIgnore: true */"process");
+    //         args = process.argv.slice(2) as string[];
+    //     } else {
+    //         args = [];
+    //     }
+    // }
 
-    mono_wasm_set_main_args(main_assembly_name, args);
-    if (runtimeHelpers.waitForDebugger == -1) {
-        mono_log_info("waiting for debugger...");
-        await mono_wasm_wait_for_debugger();
-    }
-    const method = find_entry_point(main_assembly_name);
+    // mono_wasm_set_main_args(main_assembly_name, args);
+    // if (runtimeHelpers.waitForDebugger == -1) {
+    //     mono_log_info("waiting for debugger...");
+    //     await mono_wasm_wait_for_debugger();
+    // }
+    // const method = find_entry_point(main_assembly_name);
 
-    const res = await runtimeHelpers.javaScriptExports.call_entry_point(method, args);
+    // const res = await runtimeHelpers.javaScriptExports.call_entry_point(method, args);
 
-    // one more timer loop before we return, so that any remaining queued calls could run
-    await new Promise(resolve => globalThis.setTimeout(resolve, 0));
+    // // one more timer loop before we return, so that any remaining queued calls could run
+    // await new Promise(resolve => globalThis.setTimeout(resolve, 0));
 
-    return res;
+    // return res;
+    return (Module as any)["callMain"](args);
```

Comment:
this will not work for async main

Reply:
I'm aware, temporal thing to be able to make sample that calls sync main. Proper solution is more complex